src/js/components/DateInput/DateInput.js

```diff
@@ -44,6 +46,7 @@ const DateInput = forwardRef(
       name,
       onChange,
       onFocus,
+      plain,
```

Comment:
When the DateInput is contained within a FormField it adds the plain prop which was passed down to the MaskedInput through `...rest`. Now plain is always set to true on MaskedInput and the border is on the Box container. I extracted the plain prop so that I could use it to disable the border on the Box when plain is true (See line 199).